ui/perfherder/CompareSubtestDistributionView.jsx

```diff
@@ -0,0 +1,304 @@
+      originalProject.name,
+      {
+        signature: originalSubtestSignature,
+      },
+    );
+    const {
+      suite: testSuite,
+      test: subtest,
+      name: testName,
+      platform,
+    } = seriesData[0];
+    this.setState({ testSuite, subtest, testName, platform });
+
+    this.syncGraphData(
+      originalProject.name,
+      originalRevision,
+      originalSubtestSignature,
+      true,
+    );
+    this.syncGraphData(
+      newProject.name,
+      newRevision,
+      newSubtestSignature,
+      false,
+    ).finally(() => {
+      window.document.title = `${this.state.platform}: ${this.state.testName}`;
+      this.setState({ dataLoading: false });
+    });
+  }
+
+  syncGraphData = async (
+    projectName,
+    revision,
+    subtestSignature,
+    withBaseline,
+  ) => {
+    const {
+      graphSpecsField,
+      resultSetField,
+      replicateErrorField,
+      drawingDataField,
+      targetId,
+    } = withBaseline
+      ? {
+          graphSpecsField: 'graphOneSpecs',
+          resultSetField: 'originalResultSet',
+          replicateErrorField: 'originalReplicateError',
+          drawingDataField: 'baseDrawingData',
+          targetId: CompareSubtestDistributionView.BaseTargetId,
+        }
+      : {
+          graphSpecsField: 'graphTwoSpecs',
+          resultSetField: 'newResultSet',
+          replicateErrorField: 'newReplicateError',
+          drawingDataField: 'newDrawingData',
+          targetId: CompareSubtestDistributionView.NewTargetId,
+        };
+
+    const { replicateData, drawingData } = await this.fetchReplicateGraphData(
+      projectName,
+      revision,
+      subtestSignature,
+    );
+    const { resultSet, replicateDataError } = replicateData;
+    const numRuns = drawingData ? drawingData.numRuns : 0;
+
+    this.setState({
+      [resultSetField]: resultSet,
+      [replicateErrorField]: replicateDataError,
+      [drawingDataField]: drawingData,
+      [graphSpecsField]: this.getGraphSpecs(
+        targetId,
+        numRuns,
+        replicateDataError,
+      ),
+    });
+  };
+
+  getGraphSpecs = (targetId, numRuns, error = false) =>
+    !error
+      ? {
+          ...this.state.commonGraphSpecs,
+          title: `${targetId} replicates over ${numRuns} run${
+            numRuns > 1 ? 's' : ''
+          }`,
+        }
+      : {
+          target: '',
+          data: [],
+          title: `${targetId} Replicates`,
+          chart_type: 'missing-data',
+          missing_text: 'No Data Found',
+          width: 1000,
+          height: 275,
+        };
+
+  fetchReplicateGraphData = async (project, revision, subtestSignature) => {
+    const replicateData = {};
+
+    const response = await PushModel.getList({ repo: project, revision });
+    const { results: pushPatch } = await response.json();
+    const [first] = pushPatch;
+    replicateData.resultSet = first;
+
+    const perfDatumList = await PerfSeriesModel.getSeriesData(project, {
```

Comment:
By skipping these 2 calls, you mean replace just them with the new API in an attempt to deprecate the old one? Or should this new endpoint cover PerfSeriesModel.getReplicateData() calls also?

Reply:
I think just to eliminate the usage of `getSeriesList` and `getSeriesData` here. It looks like `getSeriesList` is only being called to provide testName, platform, subtest, etc, and then in `fetchReplicateGraphData` calling `getSeriesData`. The new `/performance/summary` api would provide needed info from both queries. We can also eliminate usage of `PushModel.getList` and use the revision hash that's already available. 

If you don't feel comfortable making changes to the `/performance/summary` endpoint I can push the necessary changes on Tuesday or Wednesday. 

There's a lot of mapping over values that might not be necessary, such as with `replicatePromises` - it appears there's only one object being returned. It'd be great to look at what else can be simplified and refactored.

Reply:
Got it.

> There's a lot of mapping over values that might not be necessary, such as with `replicatePromises` - it appears there's only one object being returned. It'd be great to look at what else can be simplified and refactored.

I thought the same, but then I double checked this. `replicatePromises` can be an array of multiple objects. Bad thing about this is the page breaks because of multiple timed out queries. I think this is a issue similar to bug 1505796.

Reply:
> If you don't feel comfortable making changes to the `/performance/summary` endpoint I can push the necessary changes on Tuesday or Wednesday.

I think I managed to do the BE changes. I had to relax the query param validation a bit (make `framework` param optional).

Reply:
Looks good :)